src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -87,10 +84,11 @@ internal static IntPtr AllocateSslContext(long options, SafeX509Handle certHandl
             }
             catch
             {
-                Marshal.StructureToPtr(sslContext, sslContextPtr, false);
-                FreeSslContext(sslContextPtr);
-                return IntPtr.Zero;
+                Disconnect(sslContext.sslPtr);
+                throw;
             }
+
+            IntPtr sslContextPtr = Marshal.AllocHGlobal(Marshal.SizeOf<SslContext>());
```

Comment:
I'd previously provided feedback on this usage of SslContext as a struct, recommending instead that it be a class and that we just use a GCHandle rather than allocating native memory and marshaling the contents back and forth every time we need it.  Is that going to be done separately?